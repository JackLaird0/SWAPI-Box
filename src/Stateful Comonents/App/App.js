import React, { Component } from 'react';
import './App.css';
import TitleCategoriesContainer from './../../Stateless Components/TitleCategoriesContainer/TitleCategoriesContainer';
import ScrollAndCategoriesContainer from './../../Stateless Components/ScrollAndCategoriesContainer/ScrollAndCategoriesContainer'

class App extends Component {
  render() {
    return (
      <div className="app-component">
        <TitleCategoriesContainer />
        <ScrollAndCategoriesContainer /> 
      </div>

    )}
}

export default App;
