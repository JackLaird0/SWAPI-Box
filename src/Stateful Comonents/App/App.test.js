import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import '../../setupTests';

describe('App tests', () => {
  it('Should match the snapshot', () => {
    const appComponent = shallow(<App />, {disableLifecycleMethods: true})

    expect(appComponent).toMatchSnapshot()
  })

  it('Should render with the correct default state', () => {
    const appComponent = shallow(<App />, {disableLifecycleMethods: true})

    expect(appComponent.state('vehicles')).toEqual([])
    expect(appComponent.state('openingCrawl')).toEqual([]);
    expect(appComponent.state('planets')).toEqual([]);
    expect(appComponent.state('people')).toEqual([]);
    expect(appComponent.state('selectedCategory')).toEqual('');
    expect(appComponent.state('favorites')).toEqual([])
  })

  it('Should add to the favorites state when a favorite is saved', () => {
    const appComponent = shallow(<App />, {disableLifecycleMethods: true})
    const favoriteData = {hey: 'hello'}
    appComponent.instance().saveFavorites(favoriteData)

    expect(appComponent.state('favorites')).toEqual([favoriteData])
  })

  it('Should add to the existing state when a favorite is saved', () => {
    const appComponent = shallow(<App />, {disableLifecycleMethods: true})
    const favoriteData = {hey: 'hello'}
    const mockData = {hey: 'hello'}
    appComponent.instance().saveFavorites(favoriteData)
    appComponent.instance().saveFavorites(mockData)
    const expected = [{hey: 'hello'}, {hey: 'hello'}]

    expect(appComponent.state('favorites')).toEqual(expected)
  })
})
