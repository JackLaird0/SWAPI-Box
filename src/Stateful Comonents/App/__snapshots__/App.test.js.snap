// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TitleCategoriesContainer
          selectFavorites={[Function]}
          selectPeople={[Function]}
          selectPlanets={[Function]}
          selectVehicle={[Function]}
/>,
        <ScrollAndCategoriesContainer
          openingCrawl={Array []}
          peopleData={Array []}
          planetData={Array []}
          saveFavorite={[Function]}
          selectedCategory=""
          vehicleData={Array []}
/>,
      ],
      "className": "app-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "selectFavorites": [Function],
          "selectPeople": [Function],
          "selectPlanets": [Function],
          "selectVehicle": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "openingCrawl": Array [],
          "peopleData": Array [],
          "planetData": Array [],
          "saveFavorite": [Function],
          "selectedCategory": "",
          "vehicleData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TitleCategoriesContainer
            selectFavorites={[Function]}
            selectPeople={[Function]}
            selectPlanets={[Function]}
            selectVehicle={[Function]}
/>,
          <ScrollAndCategoriesContainer
            openingCrawl={Array []}
            peopleData={Array []}
            planetData={Array []}
            saveFavorite={[Function]}
            selectedCategory=""
            vehicleData={Array []}
/>,
        ],
        "className": "app-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "selectFavorites": [Function],
            "selectPeople": [Function],
            "selectPlanets": [Function],
            "selectVehicle": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "openingCrawl": Array [],
            "peopleData": Array [],
            "planetData": Array [],
            "saveFavorite": [Function],
            "selectedCategory": "",
            "vehicleData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
